dvtemplates/fedora_new_image.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+error_message() {
+    if [ $1 -eq 1 ]; then
+        echo -e "Unable to connect to the Fedora repository"
+    elif [ $1 -eq 2 ]; then
+        echo -e "Invalid Image Version"
+    elif [ $1 -eq 3 ]; then
+        echo -e "Unable to connect to the Fedora Image repository"
+    elif [ $1 -eq 4 ]; then
+        echo -e "Container Image for the Latest Fedora version is already present. Exiting"
+    elif [ $1 -eq 5 ]; then
+        echo -e "Unable to push the image to the local Registry."
+    elif [ $1 -eq 6 ]; then
+        echo -e "Unable to push the image to the Quay Registry."
+    elif [ $1 -eq 7 ]; then
+        echo -e "End to End Tests fail."
+    fi
+    exit $1
+}
+
+trap 'error_message $?' EXIT
+
+#Insecure registry port
+port=5000
+
+#set the final registry to hold the fedora images
+FEDORA_REPO="quay.io/kubevirt/fedora-images"
+BASE_URL=https://download.fedoraproject.org/pub/fedora/linux/releases/
+
+wget -qO index.html $BASE_URL || exit 1
+FEDORA_VERSION=`cat index.html | sed -e 's/.*>\(.*\)\/<.*/\1/' | sort -rn | head -n 1`
+re='^ *[0-9]+ *$'
+if ! [[ $FEDORA_VERSION =~ $re ]] ; then
+    error_message 2
+fi
+echo "Latest Fedora version is : ${FEDORA_VERSION}"
+
+# Fetch the old image
+docker pull -a $FEDORA_REPO
+
+image_tag=$(docker images $FEDORA_REPO --format "{{json .Tag }}" | sort -rn | head -n 1)
+FEDORA_OLD_VERSION=`echo "$image_tag" | tr -d '"'`
+echo "Fedora version in the Image Registry is : ${FEDORA_OLD_VERSION}"
+
+if [ -z "$image_tag" ]; then
+    echo -e "No Container Image found in the registry"
+fi
+
+
+if [[ "$FEDORA_VERSION" == "$FEDORA_OLD_VERSION" ]]; then
+    error_message 4
+fi
+
+IMAGE_URL=${BASE_URL}${FEDORA_VERSION}/Cloud/x86_64/images/
+wget -qO image.html $IMAGE_URL || exit 3
+FEDORA_IMAGE=$(grep Fedora.*qcow2 image.html | sed -e 's/.*>\(Fedora.*qcow2\)<.*/\1/')
+
+URL=${IMAGE_URL}${FEDORA_IMAGE}
+
+wget -q --show-progress $URL || exit 3
+echo -e "Successfully downloaded new Fedora $FEDORA_VERSION image"
+
+docker build . -t localhost:${port}/disk
+docker push localhost:${port}/disk || exit 5
+
+# Run tests
+cd ${PWD}/../;
+curl https://mirror.openshift.com/pub/openshift-v4/clients/oc/4.4/linux/oc.tar.gz | tar -C . -xzf -
```

Comment:
Do you have to download oc binary, when the test is using `KUBE_CMD=kubectl`?

Reply:
Yes, because we need to run the oc process --local` command to create and delete a VM even in a k8s cluster
https://github.com/kubevirt/common-templates/pull/307/files#diff-440ae5f3498ab37b0132430ad3c9084401907b4b0f97dab16f38f0e4c3174a2fR77
https://github.com/kubevirt/common-templates/pull/307/files#diff-440ae5f3498ab37b0132430ad3c9084401907b4b0f97dab16f38f0e4c3174a2fR107

Reply:
If you always download the `oc` because of the `oc process` requirement, then, for consistency matters, can you just use oc?

IMHO, if the `oc` command is a strong requirement, there is no need to mix `oc` and `kubectl` in the script...

Reply:
Good point, changed everything back to oc, since we "have" to use it for certain commands.